packages/prop-controllers/src/responsive-select/responsive-select.test.ts

```diff
@@ -0,0 +1,186 @@
+    })
+  })
+
+  describe('function call returns a descriptor', () => {
+    test('when called with options value', () => {
+      // Act
+      const result = ResponsiveSelect({
+        label: 'Style',
+        labelOrientation: 'horizontal',
+        options: [
+          { value: 'solid', label: 'Solid' },
+          { value: 'dashed', label: 'Dashed' },
+          { value: 'dotted', label: 'Dotted' },
+          { value: 'blended', label: 'Blended' },
+        ],
+        defaultValue: 'solid',
+      })
+
+      // Assert
+      expect(result).toEqual({
+        options: {
+          label: 'Style',
+          labelOrientation: 'horizontal',
+          options: [
+            { value: 'solid', label: 'Solid' },
+            { value: 'dashed', label: 'Dashed' },
+            { value: 'dotted', label: 'Dotted' },
+            { value: 'blended', label: 'Blended' },
+          ],
+          defaultValue: 'solid',
+        },
+        type: 'ResponsiveSelect',
+        version: 1,
+      })
+    })
+
+    test('when called with options callback', () => {
+      // Act
+      const result = ResponsiveSelect(() => ({
+        label: 'Style',
+        labelOrientation: 'horizontal',
+        options: [
+          { value: 'solid', label: 'Solid' },
+          { value: 'dashed', label: 'Dashed' },
+          { value: 'dotted', label: 'Dotted' },
+          { value: 'blended', label: 'Blended' },
+        ],
+        defaultValue: 'solid',
+      }))
+
+      // Assert
+      expect(result).toEqual({
+        options: result.options,
+        type: 'ResponsiveSelect',
+        version: 1,
+      })
+    })
+  })
+
+  describe('fromPropData', () => {
+    test('v1', () => {
+      // Arrange
+      const data: PropData<typeof ResponsiveSelect> = {
+        [ControlDataTypeKey]: ResponsiveSelect.discriminator.dataKey,
+        value: [
+          {
+            deviceId: 'desktop',
+            value: 'solid',
+          },
+        ],
+      }
+
+      // Act
+      const result = ResponsiveSelect.fromPropData(data)
+
+      // Assert
+      expect(result).toEqual(data.value)
+    })
+
+    test('v0', () => {
+      // Arrange
+      const data = [
+        {
+          deviceId: 'desktop',
+          value: 'solid',
+        },
+      ] satisfies PropData<typeof ResponsiveSelect>
+
+      // Act
+      const result = ResponsiveSelect.fromPropData(data)
+
+      // Assert
+      expect(result).toEqual(data)
+    })
+  })
+
+  describe('toPropData', () => {
+    test('returns v1 data when definition version is 1', () => {
+      // Arrange
+      const definition = {
+        type: Types.ResponsiveSelect,
+        version: 1,
+        options: {
+          options: [
+            { value: 'solid', label: 'Solid' },
+            { value: 'dashed', label: 'Dashed' },
+            { value: 'dotted', label: 'Dotted' },
+            { value: 'blended', label: 'Blended' },
+          ],
+        },
+      } satisfies Descriptor<typeof ResponsiveSelect>
```

Comment:
I'm curious about your criteria for using `satisfies`, or giving a type to the binding, or using the descriptor constructor and letting TypeScript infer it.

Reply:
I could have used a descriptor constructor here, the main reason for why the descriptor is manually spelled out here is because our tests are largely copy & pasted, and the descriptor was spelled out in the original tests that this is based on. That said, I have considered using a constructor here when I was reworking these, but we use the constructor elsewhere in these tests, and so I thought a test with an explicit descriptor literal with a `satisfies` would actually be a good thing to keep.

As far `satisfies` vs giving an explicit type to the variable, I was mostly just testing how the type is deduced. I do prefer to use `satisfies` for object literals by default.